cli/command/container/wait.go

```diff
@@ -28,6 +29,10 @@ func NewWaitCommand(dockerCli *command.DockerCli) *cobra.Command {
 			return runWait(dockerCli, &opts)
 		},
 	}
+
+	flags := cmd.Flags()
+	flags.StringVar(&opts.condition, "condition", "not-running", "Wait until a container state reaches the given condition, either 'not-running' (default), 'next-exit', or 'removed'")
```

Comment:
`docker restart ...` and restart policies technically causes an atomic `stop` followed by a `start` (with a timeout before actually starting the process in the case of restart policies).